### PR TITLE
[Metricbeat] Remove elasticsearc.index.created from the SM code

### PR DESCRIPTION
**Pull request by sayden**
## What does this PR do?

SM `index` metricset does an API HTTP call to elasticsearch to retrieve index metadata information. This information is required to populate field `elasticsearch.index.created` or (`index_stats.created` if `xpack.enabled: true` when configuring the module).

The code removes the API call to `/_cluster/state/metadata`.

## Why is it important?

The HTTP call to `/_cluster/state/metadata` can be VERY expensive. The reason for this is that it returns the entire mapping of all indices on a cluster. If a user has many indices and / or each index has many fields, this could lead to very heavy responses from elasticsearchr (20 Mb in big clusters measured with curl).

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

* This is a breaking change, because users could be using field `elasticsearch.index.created` or `index_stats.created` and they will be removed.
* Stack Monitoring UI seems to keep working without any issues. I don't see this field populated anywhere.

### CODE REVIEW

**Comment by ycombinator** [comment]
> Stack Monitoring UI seems to keep working without any issues. I don't see this field populated anywhere.

@sayden This would be my main concern with removing this field. Can we get someone from @elastic/stack-monitoring-ui team to **confirm** that removing this field will have no impact on the UI? Once we have that, I'm good with removing it to reduce the expensive API call.

**Review by jasonrhodes — APPROVED** [review]
I can confirm there are _no_ references to any field named "created" in the Stack Monitoring UI at this moment. 

For later reference, here is a search for "created" in x-pack/plugins/monitoring (excluding snapshot tests and SVG files):

Screen Shot 2021-04-19 at 12 12 47 PM


<details>
```
15 results - 10 files

x-pack/plugins/monitoring/public/plugin.ts:
   32  import { createLegacyAlertTypes } from './alerts/legacy_alert';
   33: import { createDiskUsageAlertType } from './alerts/disk_usage_alert';
   34  import { createThreadPoolRejectionsAlertType } from './alerts/thread_pool_rejections_alert';

  172      alertTypeRegistry.register(createCpuUsageAlertType());
  173:     alertTypeRegistry.register(createDiskUsageAlertType());
  174      alertTypeRegistry.register(createMemoryUsageAlertType());

x-pack/plugins/monitoring/public/alerts/disk_usage_alert/index.tsx:
  19  
  20: export function createDiskUsageAlertType(): AlertTypeModel<MonitoringAlertTypeParams> {
  21    return {

x-pack/plugins/monitoring/public/alerts/lib/alerts_toast.tsx:
  81          id="xpack.monitoring.healthCheck.disabledWatches.title"
  82:         defaultMessage="New alerts created"
  83        />

x-pack/plugins/monitoring/public/alerts/lib/get_alert_panels_by_category.test.tsx:
  48    params: {},
  49:   createdBy: null,
  50    updatedBy: null,
  51:   createdAt: new Date('2020-12-08'),
  52    updatedAt: new Date('2020-12-08'),

x-pack/plugins/monitoring/public/alerts/lib/get_alert_panels_by_node.test.tsx:
  46    params: {},
  47:   createdBy: null,
  48    updatedBy: null,
  49:   createdAt: new Date('2020-12-08'),
  50    updatedAt: new Date('2020-12-08'),

x-pack/plugins/monitoring/server/lib/mb_safe_query.ts:
  11   * queries/aggs continue to work but we need to handle the reality that these aliases will not
  12:  * exist for older metricbeat-* indices, created before the aliases existed.
  13   *

x-pack/plugins/monitoring/server/lib/details/get_series.js:
  245        console.log(
  246:         `metric.debug field=${metric.field} bucketsCreated: ${countBuckets(
  247            get(response, 'aggregations.check')

x-pack/plugins/monitoring/server/lib/elasticsearch/convert_metric_names.js:
  17   *
  18:  * Historically, the `get_nodes` function created an aggregation with multiple sub `date_histogram`
  19   * aggregations for each metric aggregation. From a top down view, the entire aggregations look liked:

x-pack/plugins/monitoring/server/lib/metrics/beats/metrics.js:
  72      description: i18n.translate('xpack.monitoring.metrics.beats.eventsRate.totalDescription', {
  73:       defaultMessage: 'All events newly created in the publishing pipeline',
  74      }),

x-pack/plugins/monitoring/server/routes/api/v1/alerts/enable.ts:
  79  
  80:         let createdAlerts: Array<SanitizedAlert<AlertTypeParams>> = [];
  81          const disabledWatcherClusterAlerts = await disableWatcherClusterAlerts(

  86          if (disabledWatcherClusterAlerts) {
  87:           createdAlerts = await Promise.all(
  88              alerts.map((alert) => alert.createIfDoesNotExist(alertsClient, actionsClient, actions))

  91  
  92:         return response.ok({ body: { createdAlerts, disabledWatcherClusterAlerts } });
  93        } catch (err) {
```

</details>

File: metricbeat/module/elasticsearch/index/data_xpack.go [hunk 1]
```diff
@@ -200,12 +194,6 @@ func addClusterStateFields(idx *Index, clusterState common.MapStr) error {
 		return errors.Wrap(err, "failed to get shards from routing table")
 	}
 
-	created, err := getIndexCreated(indexMetadata)
-	if err != nil {
-		return errors.Wrap(err, "failed to get index creation time")
-	}
-	idx.Created = created
```
Comment (ycombinator) [inline metricbeat/module/elasticsearch/index/data_xpack.go:207]: I think we can remove the `Created` field from the struct as well?

**Comment by ycombinator** [comment]
Thanks for checking and confirming, @jasonrhodes.

**Comment by ycombinator** [comment]
> The code removes the API call to /_cluster/state/metadata.

@sayden Reviewing the code changes in this PR, AFAICT we are not removing the API call itself. This PR is simply removing the code that _parses_ the cluster state API response to extract the index metadata field from it.

File: metricbeat/module/elasticsearch/index/data_xpack.go (hunk 1 above)
Comment (sayden) [reply metricbeat/module/elasticsearch/index/data_xpack.go:207]: Good point

**Comment by sayden** [comment]
> > The code removes the API call to /_cluster/state/metadata.
> 
> @sayden Reviewing the code changes in this PR, it looks like we are not actually removing the API call to ES. This PR is simply removing the code that _parses_ the cluster state API response to extract the index metadata field from it.

Yes, sorry for that. I cherry-picked the commit from a different branch and I didn't realize that I split the change in two commits

**Comment by sayden** [comment]
Ok, I actually removed the call now :sweat_smile: (actually, the request into a different call), deleted the field in struct and added a changelog entry into breaking changes.

**Review by ycombinator — APPROVED** [review]
LGTM.

**Comment by sayden** [comment]
Thank you for your help too, @jasonrhodes

**Pull request merged**